src/Microsoft.AspNet.Razor.Runtime/TagHelperContent.cs

```diff
@@ -0,0 +1,80 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+
+namespace Microsoft.AspNet.Razor.Runtime
+{
+    /// <summary>
+    /// Abstract class which is used to buffer content returned by <see cref="TagHelpers.ITagHelper"/>s.
+    /// </summary>
+    public abstract class TagHelperContent : IEnumerable<string>
+    {
+        /// <summary>
+        /// Gets a value indicating if the content was modifed.
+        /// </summary>
+        public abstract bool IsModified { get; }
+
+        /// <summary>
+        /// Gets a value indicating if the content is empty.
+        /// </summary>
+        public abstract bool IsEmpty { get; }
+
+        /// <summary>
+        /// Gets a value indicating if the content is whitespace.
+        /// </summary>
+        public abstract bool IsWhiteSpace { get; }
+
+        /// <summary>
+        /// Sets the content.
+        /// </summary>
+        /// <param name="value">The string that replaces the content.</param>
+        /// <returns>A reference to this instance after the set operation has completed.</returns>
+        public abstract TagHelperContent SetContent(string value);
+
+        /// <summary>
+        /// Sets the content by copying the contents of the specified <see cref="TagHelperContent"/>.
+        /// </summary>
+        /// <param name="tagHelperContent">The content be set.</param>
+        /// <returns>A reference to this instance after the set operation has completed.</returns>
+        public abstract TagHelperContent SetContent(TagHelperContent tagHelperContent);
+
+        /// <summary>
+        /// Appends to the existing content.
+        /// </summary>
+        /// <param name="value">The value to be appended.</param>
+        /// <returns>A reference to this instance after the append operation has completed.</returns>
+        public abstract TagHelperContent Append(string value);
+
+        /// <summary>
+        /// Appends the value from the specified <see cref="TagHelperContent"/> to the
+        /// existing <see cref="TagHelperContent"/>.
+        /// </summary>
+        /// <param name="tagHelperContent">The content to be appended.</param>
```

Comment:
- content -> `<see cref="TagHelperContent"/>`
- minor nit: all `Append()` parameters could be described as "... that is appended to the content" for increased consistency with the `SetContent()` methods